### PR TITLE
GibbsSeaWater_jll update

### PR DESCRIPTION
**Pull request by kouketsu**
Update for support platforms

### CODE REVIEW

File: G/GibbsSeaWater/build_tarballs.jl [hunk 1]
```diff
@@ -8,8 +8,7 @@ version = v"3.5.0"
 # Collection of sources required to build GibbsSeaWater
 # note 3.05.0-4 is too old to be cross-compiled on FreeBSD
 sources = [
-    "https://github.com/TEOS-10/GSW-C/archive/d392e91cb63341f543ed1609c4eff613055ab3cb.zip" =>
-    "454c46ec00468aeba048fa4f3cee095f927925ab699b1d1b3a25a838cef2d22c",
+    "https://github.com/TEOS-10/GSW-C/archive/d392e91cb63341f543ed1609c4eff613055ab3cb.zip" => "454c46ec00468aeba048fa4f3cee095f927925ab699b1d1b3a25a838cef2d22c",
```
Comment (giordano) [inline G/GibbsSeaWater/build_tarballs.jl:11]: You only changed indentation here?  Why?  Also, quoting from the [log](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=20049&view=logs&j=96c13d9d-dfd2-51e5-2f76-4a0b3ce09abc&t=09532434-8b0a-5f0e-118d-a3181cde5ec8&l=30):
```
┌ Warning: Using a pair as source is deprecated, use ArchiveSource instead
└ @ BinaryBuilderBase /depot/packages/BinaryBuilderBase/QQlL4/src/Sources.jl:278
```
```suggestion
    ArchiveSource("https://github.com/TEOS-10/GSW-C/archive/d392e91cb63341f543ed1609c4eff613055ab3cb.zip", "454c46ec00468aeba048fa4f3cee095f927925ab699b1d1b3a25a838cef2d22c"),
```

File: G/GibbsSeaWater/build_tarballs.jl [hunk 2]
```diff
@@ -35,4 +34,4 @@ dependencies = [
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.7")
```
Comment (giordano) [inline G/GibbsSeaWater/build_tarballs.jl:37]: Any specific reasons for requiring julia v1.7 and not 1.6?

Also, note that changing the julia compat bound means that you ***must*** change the version number, see #2763

**Review by giordano — CHANGES_REQUESTED** [review]

File: G/GibbsSeaWater/build_tarballs.jl [hunk 3]
```diff
@@ -8,15 +8,21 @@ version = v"3.5.0"
...
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
+export CC="cc"
+if [[ "${target}" == aarch64-apple-* ]]; then
+    export CC="clang"
+    echo $CFLAGS
+    echo $LDFLAGS
+fi
```
Comment (giordano) [inline G/GibbsSeaWater/build_tarballs.jl:22]: Nothing of this is needed
```suggestion
```
`CC` already points to `clang` for BSD systems.

File: G/GibbsSeaWater/build_tarballs.jl [hunk 4]
```diff
@@ -8,15 +8,21 @@ version = v"3.5.0"
...
 cd $WORKSPACE/srcdir/GSW-C-*
-cc $CFLAGS -fPIC -c -O3 -Wall gsw_oceanographic_toolbox.c gsw_saar.c
-cc $LDFLAGS -fPIC -shared -o libgswteos.$dlext gsw_oceanographic_toolbox.o gsw_saar.o -lm
+$CC $CFLAGS -fPIC -c -O3 -Wall gsw_oceanographic_toolbox.c gsw_saar.c
+$CC -L${libdir} $LDFLAGS -fPIC -shared -o libgswteos.$dlext gsw_oceanographic_toolbox.o gsw_saar.o -lm
 mkdir -p ${libdir}
 cp libgswteos.$dlext ${libdir}
```
Comment (giordano) [inline G/GibbsSeaWater/build_tarballs.jl:27]: Changing `cc` to `${CC}` doesn't really change anything in practice:
```suggestion
mkdir -p ${libdir}
cc -fPIC -c -O3 -Wall gsw_oceanographic_toolbox.c gsw_saar.c
cc -fPIC -shared -o "${libdir}/libgswteos.${dlext}" gsw_oceanographic_toolbox.o gsw_saar.o -lm
```

**Review by giordano — CHANGES_REQUESTED** [review]
You still haven't changed the version number, which is ***mandatory***.  Isn't there a newer upstream release to build?

**Comment by kouketsu** [comment]
I'm sorry I have took mistakes many times. Thank you for your support @giordano. 
I have bumped the version.

File: G/GibbsSeaWater/build_tarballs.jl [hunk 5]
```diff
@@ -3,20 +3,24 @@
...
 script = raw"""
 cd $WORKSPACE/srcdir/GSW-C-*
-cc $CFLAGS -fPIC -c -O3 -Wall gsw_oceanographic_toolbox.c gsw_saar.c
-cc $LDFLAGS -fPIC -shared -o libgswteos.$dlext gsw_oceanographic_toolbox.o gsw_saar.o -lm
+CC=cc
+if [[ "${target}" == aarch64-apple-* ]]; then
+    CC="gcc"
+fi
+$CC $CFLAGS -fPIC -c -O3 -Wall gsw_oceanographic_toolbox.c gsw_saar.c
+$CC $LDFLAGS -fPIC -shared -o libgswteos.$dlext gsw_oceanographic_toolbox.o gsw_saar.o -lm
```
Comment (giordano) [inline G/GibbsSeaWater/build_tarballs.jl:23]: Again, you don't need this:
```suggestion
cc -fPIC -c -O3 -Wall gsw_oceanographic_toolbox.c gsw_saar.c
cc -fPIC -shared -o libgswteos.$dlext gsw_oceanographic_toolbox.o gsw_saar.o -lm 
```

What is needed to use clang as compiler for aarch64-darwin is https://github.com/JuliaPackaging/Yggdrasil/blob/c8d4f76148c1c82667bb8f6c3b442aed57d184a8/L/LLVMOpenMP/build_tarballs.jl#L63-L65

Comment (kouketsu) [reply G/GibbsSeaWater/build_tarballs.jl]: Thank you for your advice @giordano.
I fixed it.

File: G/GibbsSeaWater/build_tarballs.jl [hunk 6]
```diff
@@ -1,22 +1,29 @@
...
 script = raw"""
+if [[ "${target}" == aarch64-apple-* ]]; then
+    # Linking libomp requires the function `__divdc3`, which is implemented in
+    # `libclang_rt.osx.a` from LLVM compiler-rt.
+    LDFLAGS="-L${libdir}/darwin -lclang_rt.osx"
+fi
 cd $WORKSPACE/srcdir/GSW-C-*
 cc $CFLAGS -fPIC -c -O3 -Wall gsw_oceanographic_toolbox.c gsw_saar.c
-cc $LDFLAGS -fPIC -shared -o libgswteos.$dlext gsw_oceanographic_toolbox.o gsw_saar.o -lm
+cc $LDFLAGS -fPIC -shared -o libgswteos.$dlext gsw_oceanographic_toolbox.o gsw_saar.o -lm
```
Comment (giordano) [inline G/GibbsSeaWater/build_tarballs.jl:26]: This introduced an extra whitespace
```suggestion
cc $LDFLAGS -fPIC -shared -o libgswteos.$dlext gsw_oceanographic_toolbox.o gsw_saar.o -lm
```

File: G/GibbsSeaWater/build_tarballs.jl [hunk 7]
```diff
@@ -1,22 +1,29 @@
...
+
 # Bash recipe for building across all platforms
 script = raw"""
+if [[ "${target}" == aarch64-apple-* ]]; then
+    # Linking libomp requires the function `__divdc3`, which is implemented in
+    # `libclang_rt.osx.a` from LLVM compiler-rt.
+    LDFLAGS="-L${libdir}/darwin -lclang_rt.osx"
+fi
 cd $WORKSPACE/srcdir/GSW-C-*
 cc $CFLAGS -fPIC -c -O3 -Wall gsw_oceanographic_toolbox.c gsw_saar.c
```
Comment (giordano) [inline G/GibbsSeaWater/build_tarballs.jl:25]: Not introduced here, but there is no `CFLAGS` variable in the environment (there used to be years ago).

File: G/GibbsSeaWater/build_tarballs.jl [hunk 8]
```diff
@@ -30,9 +37,13 @@ products = [
     LibraryProduct("libgswteos", :libgswteos)
 ]
 
+llvm_version = v"13.0.1"
 # Dependencies that must be installed before this package can be built
 dependencies = [
+    # We need libclang_rt.osx.a for linking libomp, because this library provides the
```
Comment (giordano) [inline G/GibbsSeaWater/build_tarballs.jl:43]: The comment referring libomp doesn't make much sense here since there is no no libomp.

**Review by giordano — APPROVED** [review]
Looking good now, thanks for your contribution!

**Pull request merged**